killercoda/gateway-api/assets/benchmark.sh

```diff
@@ -0,0 +1,145 @@
+
+function convert_to_ms() {
+    time_str=$1
+    min=$(echo $time_str | grep -oP '\d+(?=m)' | sed 's/^0*//')
+    sec=$(echo $time_str | grep -oP '\d+.\d+(?=s)' | sed 's/^0*//')
+    
+    if [ -z "$min" ]; then
+        min=0
+    fi
+    if [ -z "$sec" ]; then
+        sec=0
+    fi
+    
+    min_ms=$((min * 60000))
+    sec_ms=$(echo "$sec * 1000" | bc)
+    
+    total_ms=$(echo "$min_ms + $sec_ms" | bc)
+    total_ms=$(printf "%.0f" $total_ms)
+    echo $total_ms
+}
+export -f convert_to_ms
+
+function color() {
+    elapsed_ms=$1
+    if (( $(echo "$elapsed_ms > 100" | bc -l) )); then
+        color="\033[0;31m"  # Red for over 100 milliseconds
+    elif (( $(echo "$elapsed_ms >= 50" | bc -l) )); then
+        color="\033[0;33m"  # Orange for over 50 milliseconds
+    else
+        color="\033[0;32m"  # Green for under 50 milliseconds
+    fi
+    echo $color
+}
+export -f color
+
+function color_app() {
+    if [[ "$#" -eq 0 ]]; then
+        return
+    fi
+    app="${1}"
+    if [[ "$app" == "[blue]" ]]; then
+        color="\033[1;34m"
+    else
+        color="\033[1;35m"
+    fi
+    echo $color
+}
+export -f color_app
+
+(
+  echo "" > /tmp/hey.output
+  echo "" >> /tmp/hey.output
+  
+  total_count_1=0
+  total_count_2=0
+  for run in $(seq 1 $NUMBER_OF_EXECUTIONS); do
+    total_time=0
+    count_1=0
+    count_2=0
+    echo "" > /tmp/hey.output
+    echo "curl with ~10 req/sec, run $run/$NUMBER_OF_EXECUTIONS" >> /tmp/hey.output
+    for i in {1..10}; do
+      output=$( { time curl -s http://keda-meets-gw.com; } 2>&1 )
+      curl_output=$(echo "$output" | head -n -3)
+      elapsed_time=$(echo "$output" | grep real | awk '{print $2}')
+
+      elapsed_ms=$(convert_to_ms "$elapsed_time")
+      color="$(color $elapsed_ms)"
+      color_app="$(color_app $curl_output)"
+      
+      printf "${color_app}%s ${color} %5sms\033[0m\n" "$curl_output" "$elapsed_ms" >> /tmp/hey.output
+      
+      total_time=$(echo "$total_time + $elapsed_ms" | bc)
+      if [[ "$curl_output" == *"[blue]"* ]]; then
+          count_1=$((count_1 + 1))
+          total_count_1=$((total_count_1 + 1))
+      elif [[ "$curl_output" == *"[prpl]"* ]]; then
+          count_2=$((count_2 + 1))
+          total_count_2=$((total_count_2 + 1))
+      fi
+    done
+    average_time=$(echo "$total_time / 10" | bc)
+
+    color="$(color $average_time)"
+    printf "\nAverage time: ${color}%sms\033[0m\n" "$average_time" >> /tmp/hey.output
+    printf "App versions: \033[1;35m%s\033[0m / \033[1;34m%s\033[0m\n" "$count_2" "$count_1" >> /tmp/hey.output
+    if [[ "$NUMBER_OF_EXECUTIONS" -gt 1 ]]; then
+      printf "Total app versions: \033[1;35m%s\033[0m / \033[1;34m%s\033[0m\n" "$total_count_2" "$total_count_1" >> /tmp/hey.output
+    fi
+    sleep 3
+  done
+
+  echo "" >> /tmp/hey.output
+  echo "benchmark finished, press 'ctrl+c' to kill the watch loop" >> /tmp/hey.output
+  echo "when you are done observing the application scale-in" >> /tmp/hey.output
```

Comment:
```suggestion
  echo "when you are done observing the application behavior" >> /tmp/hey.output
```

This benchmark is also used when we are not doing any autoscaling  (page 3), so making this more generic.